packages/account/src/Components/forms/form-fields/__tests__/tax-identification-number.spec.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import { Formik } from 'formik';
+import { render, screen } from '@testing-library/react';
+import TaxIdentificationNumber from '../tax-indentification-number';
+
+describe('Testing <TaxIdentificationNumber/> component', () => {
+    it('should render TIN Field component', () => {
+        const props: React.ComponentProps<typeof TaxIdentificationNumber> = {
+            required: true,
+            disabled: false,
+            is_tin_popover_open: true,
+            setIsTinPopoverOpen: jest.fn(),
+            setIsTaxResidencePopoverOpen: jest.fn(),
+        };
+        render(
+            <Formik initialValues={{}} onSubmit={jest.fn()}>
+                <TaxIdentificationNumber {...props} />
+            </Formik>
+        );
+
+        expect(screen.getByText(/Tax identification number*/)).toBeInTheDocument;
+    });
+
+    it('should render TIN Field component without required', () => {
+        const props: React.ComponentProps<typeof TaxIdentificationNumber> = {
+            required: false,
+            disabled: false,
+            is_tin_popover_open: true,
+            setIsTinPopoverOpen: jest.fn(),
+            setIsTaxResidencePopoverOpen: jest.fn(),
+        };
+        render(
+            <Formik initialValues={{}} onSubmit={jest.fn()}>
+                <TaxIdentificationNumber {...props} />
+            </Formik>
+        );
+
+        expect(screen.getByText(/Tax identification number/)).toBeInTheDocument;
+    });
+
+    it('should open popover dialog when hovered', () => {
+        const props: React.ComponentProps<typeof TaxIdentificationNumber> = {
+            required: false,
+            disabled: false,
+            is_tin_popover_open: true,
+            setIsTinPopoverOpen: jest.fn(),
+            setIsTaxResidencePopoverOpen: jest.fn(),
+        };
+        render(
+            <Formik initialValues={{}} onSubmit={jest.fn()}>
+                <TaxIdentificationNumber {...props} />
+            </Formik>
+        );
+
+        const popover = screen.getByTestId('tax_identification_number_pop_over');
+        popover.dispatchEvent(new MouseEvent('click', { bubbles: true }));
```

Comment:
Can we use userEvent click here ? Instead of dispatch ?

Reply:
Yes @rupato-deriv , it works